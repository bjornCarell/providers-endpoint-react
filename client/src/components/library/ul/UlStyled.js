import styled from 'styled-components';

export const Ul = styled.ul`
  display: flex;
  align-content: ${props => {
    if (props.alignContent) return props.content;
    if (props.contentStart) return 'flex-start';
    if (props.contentEnd) return 'flex-end';
    if (props.contentCenter) return 'center';
    if (props.contentBetween) return 'space-between';
    if (props.contentAround) return 'space-around';
    return 'stretch';
  }};
  align-items: ${props => {
    if (props.alignItems) return props.alignItems;
    if (props.alignStretch) return 'stretch';
    if (props.alignEnd) return 'flex-end';
    if (props.alignCenter) return 'center';
    if (props.alignBaseLine) return 'baseline';
    return 'flex-start';
  }};
  flex-direction: ${props => (props.column ? 'column' : 'row')};
  flex-wrap: ${props => {
    if (props.wrapReverse) return 'wrap-reverse';
    if (props.noWrap) return 'nowrap';
    return 'wrap';
  }};
  justify-content: ${props => {
    if (props.justifyContent) return props.justifyContent;
    if (props.justifyCenter) return 'center';
    if (props.justifyAround) return 'space-around';
    if (props.justifyBetween) return 'space-between';
    if (props.justifyEnd) return 'flex-end';
    return 'flex-start';
  }};
  list-style-type: ${props => {
    if (props.circle) return 'circle';
    if (props.square) return 'square';
    if (props.roman) return 'upper-roman';
    if (props.alpha) return 'lower-alpha';
    return 'none';
  }};
`;
