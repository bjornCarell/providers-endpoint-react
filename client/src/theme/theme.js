export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
