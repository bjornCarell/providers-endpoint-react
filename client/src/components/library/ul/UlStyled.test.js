import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { theme } from '../../../theme/theme';
import 'jest-styled-components';
import { Ul } from './UlStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Ul theme={theme} />, div);
});

it('renders with default conditional styles', () => {
  const tree = renderer.create(<Ul theme={theme}>Providers</Ul>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'flex');
  expect(tree).toHaveStyleRule('align-content', 'stretch');
  expect(tree).toHaveStyleRule('align-items', 'flex-start');
  expect(tree).toHaveStyleRule('flex-direction', 'row');
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
  expect(tree).toHaveStyleRule('justify-content', 'flex-start');
  expect(tree).toHaveStyleRule('list-style-type', 'none');
});

//
it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Ul
        theme={theme}
        contentStart
        alignStretch
        column
        wrapReverse
        justifyCenter
        circle
      >
        Providers
      </Ul>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-start');
  expect(tree).toHaveStyleRule('align-items', 'stretch');
  expect(tree).toHaveStyleRule('flex-direction', 'column');
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap-reverse');
  expect(tree).toHaveStyleRule('justify-content', 'center');
  expect(tree).toHaveStyleRule('list-style-type', 'circle');
});

//
it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Ul theme={theme} contentEnd alignEnd noWrap justifyAround square>
        Providers
      </Ul>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-end');
  expect(tree).toHaveStyleRule('align-items', 'flex-end');
  expect(tree).toHaveStyleRule('flex-wrap', 'nowrap');
  expect(tree).toHaveStyleRule('justify-content', 'space-around');
  expect(tree).toHaveStyleRule('list-style-type', 'square');
});

//
it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Ul theme={theme} contentCenter alignCenter justifyBetween roman>
        Providers
      </Ul>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'center');
  expect(tree).toHaveStyleRule('align-items', 'center');
  expect(tree).toHaveStyleRule('justify-content', 'space-between');
  expect(tree).toHaveStyleRule('list-style-type', 'upper-roman');
});

//
it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Ul theme={theme} contentBetween alignBaseLine justifyEnd alpha>
        Providers
      </Ul>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'space-between');
  expect(tree).toHaveStyleRule('align-items', 'baseline');
  expect(tree).toHaveStyleRule('justify-content', 'flex-end');
  expect(tree).toHaveStyleRule('list-style-type', 'lower-alpha');
});

//
it('renders with set conditional styles', () => {
  const tree = renderer
    .create(
      <Ul theme={theme} contentAround>
        Providers
      </Ul>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'space-around');
});
