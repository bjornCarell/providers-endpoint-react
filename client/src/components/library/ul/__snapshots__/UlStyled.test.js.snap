// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with default conditional styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  list-style-type: none;
}

<ul
  className="c0"
>
  Providers
</ul>
`;

exports[`renders with set conditional styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style-type: circle;
}

<ul
  className="c0"
>
  Providers
</ul>
`;

exports[`renders with set conditional styles 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  list-style-type: square;
}

<ul
  className="c0"
>
  Providers
</ul>
`;

exports[`renders with set conditional styles 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  list-style-type: upper-roman;
}

<ul
  className="c0"
>
  Providers
</ul>
`;

exports[`renders with set conditional styles 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  list-style-type: lower-alpha;
}

<ul
  className="c0"
>
  Providers
</ul>
`;

exports[`renders with set conditional styles 5`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  list-style-type: none;
}

<ul
  className="c0"
>
  Providers
</ul>
`;
